### PR TITLE
QA-1595: Terraform resources for TestRunner Dashboard.

### PR DESCRIPTION
**Pull request by ichengchang**
This PR is intended to add support for Workload Identity to `TestRunner Dashboard`.
This PR is related to this PR[#529](https://github.com/broadinstitute/terra-helm/pull/529) in terra-helm.

### CODE REVIEW

File: testrunner/provider.tf
Comment (jack-r-warren) [inline testrunner/provider.tf]:
```suggestion
}

```
nit

**Review by jack-r-warren — APPROVED** [review]
This looks good--maybe we hold off to merge until we've tested in https://github.com/broadinstitute/terraform-ap-deployments/pull/434

File: testrunner/provider.tf
Comment (ichengchang) [reply testrunner/provider.tf]: Fixed.

**Comment by ichengchang** [comment]
@jack-r-warren Could you review this ticket again? Just a few changes.

Updated all tf resources with `enable` or `enable_dashboard`. 
We can just turn off `enable_dashboard` for environments that don't need dashboard resources, or vice versa.

**Review by jack-r-warren — CHANGES_REQUESTED** [review]
Three high-level things:
- When we talked on Friday Ambica was clear that preview environments were out of scope for this iteration of the project. Let's remove all references to preview environments--and frankly environment types altogether--here, since what's here is incompatible with how preview environments work. 
- There's a lot of instances here of using `count`/`element` to make multiple of a resource. Let's use `for_each` like we talked about, since it is designed for this purpose and produces better error messages if something goes wrong.
- The way you've got this set up now, you'll need multiple tfvars files for a single deployment--one with the dashboard enabled and the rest with it disabled to do permissions. A tfvars file should correspond to a single deployment of an app or service. Like we talked about, you can use submodules or for_each to set config in multiple projects from a single Terraform workspace (that is, from a single tfvars file). Let's do that to keep testrunner in line with our other Terraform patterns. (The multiple tfvars files are in https://github.com/broadinstitute/terraform-ap-deployments/pull/434/files but we don't care where you do the iteration there or here)

**Review by jack-r-warren — APPROVED** [review]
Approving to unblock

**Pull request merged**
